src/components/challenges/active.tsx

```diff
@@ -27,26 +29,46 @@ type ActiveProps = {
     }[];
     is_active: boolean;
     is_user_subscribed: number;
-  };
+  },
+  userId: string
 };
 
-const Active: FC<ActiveProps> = ({ content }) => {
+const { SUCCESS, ERROR } = ToastTypes;
+
+const Active: FC<ActiveProps> = ({ content, userId }) => {
   const [isUserSubscribed, setUserSubscribed] = useState(content.is_user_subscribed);
 
+  const subscibeUser = async () => {
+    const data = {
+      challenge_id: content.id,
+      user_id: userId,
+    };
+    try {
+      const response = await fetch(SUBSCRIBE_TO_CHALLENGE_URL, {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify(data),
+        credentials: 'include',
+      });
+      const { isUserSubscribed: subscription } = await response.json();
+      setUserSubscribed(subscription);
+      toast(SUCCESS, 'You have subscribed to the challenge');
+    } catch (error:any) {
+      toast(ERROR, error.message);
```

Comment:
If an error has `response` present, please toast the `error.response.message` which is the message returned from the RDS backend.